tests/phpunit/Unit/mw.smw.info.tests.lua

```diff
@@ -0,0 +1,44 @@
+--[[
+	Tests for smw.info module
+
+	@since 1.0
+
+	@licence GNU GPL v2+
+	@author oetterer
+]]
+
+local testframework = require 'Module:TestFramework'
+
+-- Tests
+local tests = {
+	{ name = 'info (no argument)', func = mw.smw.info,
+		args = { '' },
+		expect = { nil }
+	},
+	{ name = 'info (text, no icon)',
+		func = function ( args )
+			local ret = mw.smw.info( args )
+			return type(ret), #ret > 0
+		end,
+		args = { 'Infotext' },
+		expect = { 'string', true }
```

Comment:
There actually is very little lua code to test. All there is in lua are these trivial bindings to PHP code. It's be *nice* to test those as well (including that transformation from PHP to lua happens as expected), though by far not as critical as the actual logic, which resides in PHP. To test that logic, you can write simple unit tests in PHP, that pass in arguments to the PHP functions and assert they get the expected output. Doing that via lua falls under the "testing through the presentation layer" anti-pattern.